modules/diskq/qdisk.h

```diff
@@ -41,6 +41,7 @@ typedef struct _QDisk QDisk;
 
 QDisk *qdisk_new(void);
```

Comment:
What about adding an extra argument to qdisk_init() ?
That is already a constructor to QDisk

Reply:
I tried, but in logqueue-disk.c: disk_init() is also used in _restart_diskq. In that occurrence it would be necessary to steal the magic id from the already created disk queue beforehand, that would be rewritten back to the disk. That would look strange from initialization perspective. Also, in that module QDisk is forward declared, so it would be necessary to add a getter to the api for the magic id. So in the end, my setter function would be replaced with a getter function, so the api complexity would not change much. I would leave as it is.

Reply:
Do you mean "as it is", by leaving the `set_file_id()` function?

Reply:
I've got an idea: the problem is LogQueueDisk wants to restart, but does not know how, because the restart depends on the type of the QDisk. But this can be resolved, if I add a `restart` virtual function to QDisk. With that, we would not need neither the setter and the getter, and qdisk_init could be leaned towards a true init function of qdisk by extending the parameters as you suggested. I'll try to implement this, and come back with a patch.